gptqmodel/utils/device.py

```diff
@@ -33,8 +40,12 @@ def get_device(obj: torch.Tensor | nn.Module) -> torch.device:
         return params[0].device
     elif len(buffers) > 0:
         return buffers[0].device
-    else:
-        return CPU
+    
+    if isinstance(obj, (HookedLinear, HookedConv1D, HookedConv1d, HookedConv2d, HookedTransformerConv1D)):
+        if hasattr(obj, 'weight') and isinstance(obj.weight, torch.Tensor):
+            return obj.weight.device
+            
+    return CPU
```

Comment:
Did you have to add this to bypass some crash? Do you remember why this is needed?

Reply:
Ok. We will keep this as this maybe a bug fix.